src/realizations/catchment/Bmi_C_Adapter.cpp

```diff
@@ -402,6 +402,54 @@ std::string Bmi_C_Adapter::GetVarUnits(std::string name) {
     return units_str;
 }
 
+std::string Bmi_C_Adapter::GetVarLocation(std::string name) {
+    char location_c_str[BMI_MAX_UNITS_NAME];
+    int success = bmi_model->get_var_location(bmi_model.get(), name.c_str(), location_c_str);
+    if (success != BMI_SUCCESS) {
+        throw std::runtime_error(model_name + " failed to get variable location for " + name + ".");
+    }
+    std::string location_str(location_c_str);
```

Comment:
IIRC you could probably simplify this to just `return std::string( location_c_str )`

Reply:
ok - this works